includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -182,19 +182,19 @@ public static function has_delayed_payment( WC_Subscription $subscription ) {
 	 * @return WC_Subscription|bool The WC subscription or false if it can't be found.
 	 */
 	public static function get_subscription_from_wcpay_subscription_id( string $wcpay_subscription_id ) {
-		global $wpdb;
-
-		$subscription_id = $wpdb->get_var(
-			$wpdb->prepare(
-				"SELECT post_id FROM {$wpdb->prefix}postmeta
-				WHERE meta_key = %s
-				AND meta_value = %s",
-				self::SUBSCRIPTION_ID_META_KEY,
-				$wcpay_subscription_id
-			)
+		$subscriptions = wcs_get_subscriptions(
+			[
+				'limit'      => 1,
```

Comment:
```suggestion
				'subscriptions_per_page' => 1,
```

`wcs_get_subscriptions()` doesn't use the limit `arg`.
This is minor as we only ever expect one subscription to have the meta key